src/jshint.js

```diff
@@ -4247,12 +4305,12 @@ var JSHINT = (function() {
     x.lbp = 25;
   }(prefix("yield", function() {
     var prev = state.tokens.prev;
-    if (state.inESNext(true) && !state.funct["(generator)"]) {
+    if (state.inES6(true) && !state.funct["(generator)"]) {
```

Comment:
Done, but I've not yet documented `esversion: 7` becasue of https://github.com/jshint/jshint/issues/1939#issuecomment-62282120